### PR TITLE
[WASM, Kotlin]: Fixed methods and properties in enums

### PR DESCRIPTION
**Pull request by MaximPestryakov**
- Use static method for enums' methods and properties
- Added `BitcoinSigHashTypeExt` and `HDVersionExt`
- Fixed Kotlin JS implementation for enums
- Added tests for `BitcoinSigHashTypeExt` and `HDVersionExt`

### CODE REVIEW

**Review by Milerius — APPROVED** [review]
I'm ok to incorporate those changes, but I thought you wanted to generate those coin extension by modifying the codegen tool, if static functions for ext is ok for end-users that consume the library, that's fine by me.

**Comment by MaximPestryakov** [comment]
> I'm ok to incorporate those changes, but I thought you wanted to generate those enum extension by modifying the codegen tool, if static functions for ext is ok for end-users that consume the library, that's fine by me.

I was trying to implement it using codegen but I ran into some difficulties. So, for now `...Ext` is enough

**Review by Milerius — CHANGES_REQUESTED** [review]
Please revert the change for `package-lock.json`

File: wasm/src/CoinTypeExt.h [hunk 1]
```diff
@@ -0,0 +1,42 @@
+#pragma once
```
[the remark this answers is not in the document]
Comment (MaximPestryakov) [reply wasm/src/CoinTypeExt.h:7]: Done

File: wasm/src/CoinTypeExt.cpp [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+            .class_function("xprvVersion", &CoinTypeExt::xprvVersion, allow_raw_pointers())
+            .class_function("hrp", &CoinTypeExt::hrp, allow_raw_pointers())
+            .class_function("p2pkhPrefix", &CoinTypeExt::p2pkhPrefix, allow_raw_pointers())
+            .class_function("p2shPrefix", &CoinTypeExt::p2shPrefix, allow_raw_pointers())
+            .class_function("staticPrefix", &CoinTypeExt::staticPrefix, allow_raw_pointers())
+            .class_function("chainId", &CoinTypeExt::chainId, allow_raw_pointers())
+            .class_function("slip44Id", &CoinTypeExt::slip44Id, allow_raw_pointers())
+            .class_function("ss58Prefix", &CoinTypeExt::ss58Prefix, allow_raw_pointers())
+            .class_function("publicKeyType", &CoinTypeExt::publicKeyType, allow_raw_pointers())
+            .class_function("validate", &CoinTypeExt::validate, allow_raw_pointers())
```
[the remark this answers is not in the document]
Comment (MaximPestryakov) [reply wasm/src/CoinTypeExt.cpp]: Done

File: wasm/src/BitcoinSigHashTypeExt.cpp [hunk 3]
```diff
@@ -0,0 +1,23 @@
+#include <TrustWalletCore/TWBitcoinSigHashType.h>
```
[the remark this answers is not in the document]
Comment (MaximPestryakov) [reply wasm/src/BitcoinSigHashTypeExt.cpp:7]: Done

**Comment by Milerius** [comment]
Please @MaximPestryakov revert the package lock json change

**Comment by MaximPestryakov** [comment]
> Please @MaximPestryakov revert the package lock json change

Done

**Review by hewigovens — APPROVED** [review]

**Review by Milerius — APPROVED** [review]

**Pull request merged**